crates/stackable-versioned-macros/src/lib.rs

```diff
@@ -8,50 +8,95 @@ mod attrs;
 mod codegen;
 mod consts;
 
-/// This macro enables generating versioned structs.
+/// This macro enables generating versioned structs and enums.
 ///
-/// ## Usage Guide
+/// # Usage Guide
 ///
-/// ### Quickstart
+/// In this guide, code blocks usually come in pairs. The first code block
+/// describes how the macro is used. The second expandable block displays the
+/// generated piece of code for explanation purposes. It should be noted, that
+/// the exact code can diverge from what is being depicted in this guide. For
+/// example, `#[automatically_derived]` and `#[allow(deprecated)]` are removed
+/// in most examples to reduce visual clutter.
+///
+/// ## Declaring Versions
+///
+/// It is **important** to note that this macro must be placed before any other
+/// (derive) macros and attributes. This ensures that the macros and attributes
+/// are applied to the generated versioned instances of the struct or enum.
+///
+/// Before any of the fields or variants can be versioned, versions need to be
+/// declared at the container level. Each version currently supports two
+/// parameters: `name` and the `deprecated` flag. The `name` must be a valid
+/// (and supported) format - currently, only Kubernetes API versions are
+/// supported. The macro checks each declared version and reports any error
+/// encountered during parsing.
```

Comment:
As per https://github.com/rust-lang/rust/issues/79710#issuecomment-1868261223, we can make this a warning.

```suggestion
/// (and supported) format.
///
/// <div class="warning">
/// Currently, only Kubernetes API versions are
/// supported. The macro checks each declared version and reports any error
/// encountered during parsing.
/// </div>
```